src/System.Linq.Parallel/src/System/Linq/Parallel/QueryOperators/Inlined/NullableDecimalAverageAggregationOperator.cs

```diff
@@ -118,8 +118,11 @@ protected override bool MoveNextCore(ref Pair<decimal, long> currentElement)
 
                     if (current.HasValue)
                     {
-                        sum += current.GetValueOrDefault();
-                        count++;
+                        checked
+                        {
+                            sum += current.GetValueOrDefault();
+                            count++;
+                        }
```

Comment:
> I didn't think checked worked on decimal? 

Sorry I was unclear... I was just referring to all of the tests together and happened to comment on that one. (Decimal itself does overflow checking, regardless of whether you use checked.)

> The keyword also doesn't appear to propagate into method calls either

That's correct; it purely affects the IL contained within the checked block in that method.

> If you mean for nullable int, it lofts to long for the intermediate sum, so wont overflow before hitting the iteration limit anyways.

Ah, you're right.  That's a good point.

> For nullable long it's possible, but the current behavior is non-deterministic; the worker threads used non-checked, while the final result uses checked, which means whether you get an exception or an incorrect result depends on where the partitioner split your values

Ok, that's reasonable.  Thanks.